database/tests/desktop/core/sync_point_spec_test.cc

```diff
@@ -0,0 +1,837 @@
+    Variant index =
+        bound->index()
+            ? util::FlexbufferToVariant(bound->index_flexbuffer_root())
+            : Variant::Null();
+    query.reset(query->EqualTo(index, name));
+  }
+  if (query_params->limitToFirst()) {
+    query.reset(query->LimitToFirst(query_params->limitToFirst()));
+  }
+  if (query_params->limitToLast()) {
+    query.reset(query->LimitToLast(query_params->limitToLast()));
+  }
+  return query;
+}
+
+Event ParseEvent(const test_data::Event* event_spec, Path path) {
+  EventType type = static_cast<EventType>(event_spec->type());
+
+  EXPECT_NE(event_spec->path(), nullptr);
+  path = path.GetChild(event_spec->path()->c_str());
+  if (event_spec->name() != nullptr) {
+    path = path.GetChild(event_spec->name()->c_str());
+  }
+  Variant data = event_spec->data()
+                     ? FlexbufferToVariant(event_spec->data_flexbuffer_root())
+                     : Variant::Null();
+  DataSnapshotInternal snapshot(nullptr, data, QuerySpec(path));
+
+  const char* prev_name =
+      event_spec->prevName() != nullptr ? event_spec->prevName()->c_str() : "";
+  return Event(type, nullptr, snapshot, prev_name);
+}
+
+void SyncTreeTest::runTest(const test_data::TestCase* test_spec,
+                           Path base_path) {
+  logger_.LogInfo("Running \"%s\"", test_spec->name()->c_str());
+
+  int current_write_id = 0;
+
+  std::map<int, EventRegistration*> registrations;
+  for (const test_data::Step* spec : *test_spec->steps()) {
+    if (spec->comment()) {
+      logger_.LogInfo(" > %s", spec->comment()->c_str());
+    }
+    const char* path_str = spec->path() ? spec->path()->c_str() : "";
+    Path path = base_path.GetChild(path_str);
+    DatabaseReferenceInternal reference(nullptr, path);
+    std::vector<Event> expected;
+    for (const auto* event_spec : *spec->events()) {
+      expected.push_back(ParseEvent(event_spec, base_path));
+    }
+    switch (spec->type()) {
+      case test_data::StepType_listen: {
+        UniquePtr<QueryInternal> query(new QueryInternal(reference));
+        if (spec->params()) {
+          query = ParseQuery(std::move(query), spec->params());
+        }
+        EventRegistration* event_registration;
+        int callback_id = spec->callbackId();
+        if (callback_id != 0 && registrations.count(callback_id) != 0) {
+          event_registration = registrations[callback_id];
+        } else {
+          event_registration = new TestEventRegistration(query->query_spec());
+          if (callback_id != 0) {
+            registrations[callback_id] = event_registration;
+          }
+        }
+        std::vector<Event> actual = sync_tree_->AddEventRegistration(
+            UniquePtr<EventRegistration>(event_registration));
+        EXPECT_THAT(actual, Pointwise(EventEq(), expected));
+        break;
+      }
+      case test_data::StepType_unlisten: {
+        EventRegistration* event_registration = nullptr;
```

Comment:
This is bit inconsistent.  Above the event_registration isn't deliberately initialized.  Maybe add a nullptr assignment above as well?

Reply:
Done. I don't think it matters much as in both cases its guaranteed to be registered by the time it's used, but consistency is good.